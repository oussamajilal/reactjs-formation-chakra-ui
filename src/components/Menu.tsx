import * as React from "react"
import { Heading, ListItem, UnorderedList } from "@chakra-ui/layout";
import { Box } from "@chakra-ui/react";
import Recipe from "./Recipe";

const Menu = () => <Box>
    <Heading>My Menu</Heading>
    <UnorderedList>
        <ListItem>
            <Recipe id={1} />
        </ListItem>
        <ListItem>
            <Recipe id={2} />
        </ListItem>
        <ListItem>
            <Recipe id={3} />
        </ListItem>
    </UnorderedList>
</Box>

export default Menu;