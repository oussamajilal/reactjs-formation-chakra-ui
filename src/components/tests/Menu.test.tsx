import React from "react"
import { cleanup } from '@testing-library/react'
import { render } from "../../test-utils"
import Menu from "../Menu"

afterEach(cleanup)

describe('Menu', () => {
    it('should render correctly', () => {
        const { asFragment } = render(<Menu />)
        expect(asFragment()).toMatchSnapshot()
    })
})