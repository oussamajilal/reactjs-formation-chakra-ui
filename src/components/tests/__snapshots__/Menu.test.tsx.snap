// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render correctly 1`] = `
<DocumentFragment>
  <div
    class="css-0"
  >
    <h2
      class="chakra-heading css-1dklj6k"
    >
      My Menu
    </h2>
    <ul
      class="css-tu0njr"
      role="list"
    >
      <li
        class="css-0"
      >
        <div
          class="css-0"
        >
          <span>
            My Recipe N° 1 incremented 0 times
          </span>
          <button
            class="chakra-button css-taj3dd"
            type="button"
          >
            +
          </button>
        </div>
      </li>
      <li
        class="css-0"
      >
        <div
          class="css-0"
        >
          <span>
            My Recipe N° 2 incremented 0 times
          </span>
          <button
            class="chakra-button css-taj3dd"
            type="button"
          >
            +
          </button>
        </div>
      </li>
      <li
        class="css-0"
      >
        <div
          class="css-0"
        >
          <span>
            My Recipe N° 3 incremented 0 times
          </span>
          <button
            class="chakra-button css-taj3dd"
            type="button"
          >
            +
          </button>
        </div>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
